menu.js

```diff
@@ -394,21 +394,35 @@ projectDetails.forEach((n) => n.addEventListener('click', displayMobileCard));
 projectDetails.forEach((n) => n.addEventListener('click', displayDesktopCard));
 
 // form Validation
-const fullName = document.getElementById('footer-name');
-const email = document.getElementById('email');
-const userMessage = document.getElementById('message');
+const fullName = document.getElementById('#name');
+const email = document.getElementById('#email');
+const userMessage = document.getElementById('#message');
 const contactForm = document.querySelector('#form');
 const formInput = document.querySelector('.form-btn');
 const error = document.querySelector('.error');
 
-contactForm.addEventListener('submit', (e) => {
-  const emailValue = email.value;
-  let flag = 0;
-  for (let i = 0; i < emailValue.length; i += 1) {
-    if (emailValue[i] !== emailValue[i].toLowerCase()) {
-      e.preventDefault();
-      error.innerText = 'Please enter email in lowercase';
-      flag = 1;
-    }
-  }
+// contactForm.addEventListener('submit', (e) => {
+//   const emailValue = email.value;
+//   let flag = 0;
+//   for (let i = 0; i < emailValue.length; i += 1) {
+//     if (emailValue[i] !== emailValue[i].toLowerCase()) {
+//       e.preventDefault();
+//       error.innerText = 'Please enter email in lowercase';
+//       flag = 1;
+//     }
+//   }
+// });
+
+formInput.addEventListener('input', () => {
+  const Data = {
+    FullName: document.querySelector('#name').value,
+    EmailAdress: document.querySelector('#email').value,
+    TextArea: document.querySelector('#message').value,
+  };
+  localStorage.setItem('LocalFormStorage', JSON.stringify(Data));
 });
+
+const StoredData = JSON.parse(localStorage.getItem('LocalFormStorage'));
+document.querySelector('#name').value = StoredData.FullName;
+document.querySelector('#email').value = StoredData.EmailAdress;
+document.querySelector('#message').value = StoredData.TextArea;
```

Comment:
- Dear @hiwina, I see you use an `input` event on the button, Also this is not working to save data to local storage because of the wrong event listener for a button. please make sure the add the appropriate event listeners to preserve data and load data on reload. 
### Majore detail to make sure 
 - [ ] 1 The contact form information is persistent to page reloads
 - [ ] 2 After filling the form, and not submitting it, the entered values should be saved in the browser
 - [ ] 3 When visiting the page the form inputs should be contained, be auto-populated with, the values entered in the previous visit to the page.
 - [ ] 4 If at the previous visit to the page, in the same browser, no input was entered in the form then the form inputs should remain empty.